File: src/main/tools/system/stats_tool.js
/**
 * System Stats Tool - Node.js implementation using systeminformation
 */

const si = require('systeminformation');

/**
 * Get system statistics
 */
async function getSystemStats(args, skipAutoCapture) {
  try {
    const [cpu, mem, battery] = await Promise.all([
      si.currentLoad(),
      si.mem(),
      si.battery().catch(() => null), // Battery may not be available
    ]);

    const stats = {
      cpu_percent: cpu.currentLoad || 0,
      memory_percent: (mem.used / mem.total) * 100,
      battery_percent: battery ? battery.percent : null,
      battery_charging: battery ? battery.isCharging : null,
    };

    const content = JSON.stringify(stats, null, 2);

    return {
      success: true,
      data: {
        stats,
        llm_content: content,
      },
    };
  } catch (error) {
    console.error(`[StatsTool] Error: ${error.message}`, error);
    return { success: false, error: `Failed to get system stats: ${error.message}` };
  }
}

module.exports = {
  getSystemStats,
};

File: src/main/python/tools/system/stats_tool.py
"""
System Stats Tool - Python implementation using psutil.
"""

import logging
from typing import Dict, Any

from core.system_metrics import collect_system_stats
from tools.result import ToolResult

logger = logging.getLogger(__name__)


async def get_system_stats(args: Dict[str, Any]) -> ToolResult:
    """
    Get system statistics.
    
    Args:
        args: Dictionary (unused, but kept for interface consistency)
        
    Returns:
        ToolResult with system stats data
    """
    try:
        stats = await collect_system_stats()
        
        import json
        content = json.dumps(stats, indent=2)
        
        return ToolResult.success_result(
            {
                "stats": stats,
                "output": content,
            }
        )
    except ImportError:
        logger.error("psutil not available, cannot get system stats")
        return ToolResult.error_result("psutil library not available")
    except Exception as e:
        logger.error(f"Error getting system stats: {e}", exc_info=True)
        return ToolResult.error_result(f"Failed to get system stats: {str(e)}")
